pkg/token/identitytoken/identitytoken_test.go

```diff
@@ -0,0 +1,190 @@
+
+        n = copy(p, b.body[b.readIndex:])
+        b.readIndex += n
+
+        return n, nil
+}
+
+func (h *testHTTPClient) Do(req *http.Request) (*http.Response, error) {
+        bytes, err := json.Marshal(h.body)
+        if err != nil {
+                return nil, err
+        }
+
+        body := &bodyReadCloser{body: bytes}
+
+        return &http.Response{StatusCode: h.statusCode, Body: body}, nil
+}
+
+func createTestClient(identityServiceURL string, statusCode int, token TokenResponse) *Client {
+        c := New(identityServiceURL)
+        c.httpClient = &testHTTPClient{
+                statusCode: statusCode,
+                body:       token,
+        }
+
+        return c
+}
+
+func TestGenerateToken(t *testing.T) {
+        t.Parallel()
+        testcases := []struct {
+                name       string
+                ctx        context.Context
+                url        string
+                statusCode int
+                token      TokenResponse
+                err        error
+        }{
+                {
+                        name:       "success",
+                        ctx:        context.Background(),
+                        url:        "https://client.greenlake.hpe.com/api/iam/identity",
+                        statusCode: http.StatusOK,
+                        token: TokenResponse{
+                                AccessToken: "access-token",
+                        },
+                },
+                {
+                        name:       "status code 404",
+                        url:        "https://client.greenlake.hpe.com/api/iam/identity",
+                        ctx:        context.Background(),
+                        statusCode: http.StatusNotFound,
+                        err:        errors.New("Unexpected status code 404"),
+                },
+                {
+                        name: "no context",
+                        url:  "https://client.greenlake.hpe.com/api/iam/identity",
+                        ctx:  nil,
+                        err:  errors.New("net/http: nil Context"),
+                },
+                {
+                        name:       "status code 400",
+                        url:        "https://client.greenlake.hpe.com/api/iam/identity",
+                        ctx:        context.Background(),
+                        statusCode: http.StatusBadRequest,
+                        err:        errors.New("Bad request: {\"token_type\":\"\",\"access_token\":\"\",\"refresh_token\":\"\",\"expiry\":\"0001-01-01T00:00:00Z\",\"expires_in\":0,\"scope\":\"\",\"accessTokenOnly\":false}"),
+                },
+                {
+                        name:       "status code 401",
+                        url:        "https://client.greenlake.hpe.com/api/iam/identity",
+                        ctx:        context.Background(),
+                        statusCode: http.StatusUnauthorized,
+                        err:        errors.New("Forbidden: "),
```

Comment:
Forbidden is usually associated with 403 rather than 401.

Reply:
I added a test for both Forbidden and Unauthorized Access error code.